### ISSUE TITLE
TypeError in schedy 0.8.0

### ISSUE THREAD

**Issue by bachya** [efficiosoft/hass-apps#47]
I am using AppDaemon 4.0.1 and have this Schedy app:

```yaml
outdoor_lights_schedule:
  module: hass_apps_loader
  class: SchedyApp
  constrain_input_boolean: input_boolean.outdoor_lights_schedule_automation

  actor_type: switch

  expression_environment: |
    def is_sun_down():
      return state("sun.sun") == "below_horizon"

  rooms:
    backyard:
      actors:
        switch.backyard_lights:

      rescheduling_delay: 0

      schedule:
        - v: "on"
          rules:
            - x: "Next() if is_sun_down() else Break()"
            - {start: "15:00", end: "21:30", weekdays: "1-4, 7"}
            - {start: "15:00", end: "23:00", weekdays: 5-6}
        - v: "off"

    front_and_kitchen_patios:
      actors:
        group.patio_lights:

      rescheduling_delay: 0

      schedule:
        - v: "on"
          rules:
            - x: "Next() if is_sun_down() else Break()"
            - {start: "15:00", end: "23:00"}
        - v: "off"

  watched_entities:
    - sun.sun
```

As of Schedy 0.8.0, I've started receiving this error:

```
2020-02-03 21:30:00.062722 WARNING outdoor_lights_schedule: ------------------------------------------------------------
2020-02-03 21:30:00.062957 WARNING outdoor_lights_schedule: Unexpected error in worker for App outdoor_lights_schedule:
2020-02-03 21:30:00.063146 WARNING outdoor_lights_schedule: Worker Ags: {'id': '2aa11e2fd41c40a39975a5b0dee4a9d6', 'name': 'outdoor_lights_schedule', 'objectid': 'efcac60078ea4636ba00bfe30fe63203', 'type': 'scheduler', 'function': <bound method Room._scheduling_timer_cb of <Room R:backyard>>, 'pin_app': True, 'pin_thread': 12, 'kwargs': {'interval': 86400, '__thread_id': 'thread-12'}}
2020-02-03 21:30:00.063308 WARNING outdoor_lights_schedule: ------------------------------------------------------------
2020-02-03 21:30:00.063639 WARNING outdoor_lights_schedule: Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/appdaemon/threading.py", line 766, in worker
    funcref(self.AD.sched.sanitize_timer_kwargs(app, args["kwargs"]))
  File "/usr/local/lib/python3.8/site-packages/hass_apps/schedy/room.py", line 50, in wrapper
    result = handler(self, *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/hass_apps/schedy/room.py", line 204, in _scheduling_timer_cb
    self.apply_schedule()
  File "/usr/local/lib/python3.8/site-packages/hass_apps/schedy/room.py", line 50, in wrapper
    result = handler(self, *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/hass_apps/schedy/room.py", line 409, in apply_schedule
    self.set_value(new_scheduled_value, force_resend=force_resend)
  File "/usr/local/lib/python3.8/site-packages/hass_apps/schedy/room.py", line 587, in set_value
    changed |= actor.set_value(value, force_resend=force_resend)[0]
  File "/usr/local/lib/python3.8/site-packages/hass_apps/schedy/actor/base.py", line 285, in set_value
    self._resending_cb({"left_tries": self.cfg["send_retries"] + 1})
  File "/usr/local/lib/python3.8/site-packages/hass_apps/schedy/room.py", line 50, in wrapper
    result = handler(self, *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/hass_apps/schedy/actor/base.py", line 93, in _resending_cb
    self.do_send()
  File "/usr/local/lib/python3.8/site-packages/hass_apps/schedy/actor/generic2.py", line 139, in do_send
    self._populate_service_data(data, fmt)
  File "/usr/local/lib/python3.8/site-packages/hass_apps/schedy/actor/generic2.py", line 89, in _populate_service_data
    memo = {data}  # type: T.Set[T.Union[T.Dict, T.List]]
TypeError: unhashable type: 'dict'

2020-02-03 21:30:00.063803 WARNING outdoor_lights_schedule: ------------------------------------------------------------
```

Pinning Schedy to 0.7.0 causes the error to go away.

**Comment by adandl** [comment]
Hi - I have the same error - however found out that it has to be related to hassio 32bit (running on raspberry pi 4). 
I have a test system running under docker on my Synology where I run hassio 64bit and with the same configuration I don't see this issue...

**Comment by efficiosoft** [comment]
Hi,

Thanks for reporting.

Would you please try out the development version (from master branch) and confirm that it's fixed there?

Best regards
Robert

**Comment by adandl** [comment]
Hi Robert!
Can you please tell me how to specify to load the latest develpment version?
Thanks a lot!

**Comment by efficiosoft** [comment]
Sure, see [here](https://hass-apps.readthedocs.io/en/stable/getting-started.html) and choose to install the development version (option c).

**Comment by adandl** [comment]
Hi,

sorry - but it seems that the "requirements.txt" file is ignored - at least I still have the same issue.
Can you check in my log-file if it actually uses the development branche?

`-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] appdaemon.sh: executing... 
Looking in links: https://wheels.hass.io/alpine-3.11/armv7/
Collecting hass-apps
  Downloading https://files.pythonhosted.org/packages/39/be/10274d816dd00defa45785acf8ef20c4a7fa9cbdf10d55b8e84792c94530/hass_apps-0.20200203.0-py3-none-any.whl (54kB)
Requirement already satisfied: voluptuous>=0.11 in /usr/lib/python3.8/site-packages (from hass-apps) (0.11.5)
Collecting observable>=1.0 (from hass-apps)
  Downloading https://files.pythonhosted.org/packages/45/7c/b4b63f447378e8a0ebcd338d90f9389f57fb23253127425beacf0129edcb/observable-1.0.3-py2.py3-none-any.whl
Requirement already satisfied: appdaemon>=3.0 in /usr/lib/python3.8/site-packages (from hass-apps) (4.0.1)
Collecting cached-property>=1.5.1 (from hass-apps)
  Downloading https://files.pythonhosted.org/packages/3b/86/85c1be2e8db9e13ef9a350aecd6dea292bd612fa288c2f40d035bb750ded/cached_property-1.5.1-py2.py3-none-any.whl
Requirement already satisfied: deepdiff in /usr/lib/python3.8/site-packages (from appdaemon>=3.0->hass-apps) (4.0.9)
Requirement already satisfied: requests>=2.6.0 in /usr/lib/python3.8/site-packages (from appdaemon>=3.0->hass-apps) (2.22.0)
Requirement already satisfied: iso8601 in /usr/lib/python3.8/site-packages (from appdaemon>=3.0->hass-apps) (0.1.12)
Requirement already satisfied: Jinja2==2.10.1 in /usr/lib/python3.8/site-packages (from appdaemon>=3.0->hass-apps) (2.10.1)
Requirement already satisfied: feedparser in /usr/lib/python3.8/site-packages (from appdaemon>=3.0->hass-apps) (5.2.1)
Requirement already satisfied: aiohttp-jinja2==0.15.0 in /usr/lib/python3.8/site-packages (from appdaemon>=3.0->hass-apps) (0.15.0)
Requirement already satisfied: websocket-client in /usr/lib/python3.8/site-packages (from appdaemon>=3.0->hass-apps) (0.57.0)
Requirement already satisfied: aiohttp==3.6.2 in /usr/lib/python3.8/site-packages (from appdaemon>=3.0->hass-apps) (3.6.2)
Requirement already satisfied: python-dateutil in /usr/lib/python3.8/site-packages (from appdaemon>=3.0->hass-apps) (2.8.1)
Requirement already satisfied: python-socketio in /usr/lib/python3.8/site-packages (from appdaemon>=3.0->hass-apps) (4.4.0)
Requirement already satisfied: daemonize in /usr/lib/python3.8/site-packages (from appdaemon>=3.0->hass-apps) (2.5.0)
Requirement already satisfied: pid in /usr/lib/python3.8/site-packages (from appdaemon>=3.0->hass-apps) (2.2.5)
Requirement already satisfied: bcrypt==3.1.4 in /usr/lib/python3.8/site-packages (from appdaemon>=3.0->hass-apps) (3.1.4)
Requirement already satisfied: paho-mqtt in /usr/lib/python3.8/site-packages (from appdaemon>=3.0->hass-apps) (1.5.0)
Requirement already satisfied: yarl==1.1.0 in /usr/lib/python3.8/site-packages (from appdaemon>=3.0->hass-apps) (1.1.0)
Requirement already satisfied: pytz in /usr/lib/python3.8/site-packages (from appdaemon>=3.0->hass-apps) (2019.3)
Requirement already satisfied: pyyaml==5.1 in /usr/lib/python3.8/site-packages (from appdaemon>=3.0->hass-apps) (5.1)
Requirement already satisfied: astral in /usr/lib/python3.8/site-packages (from appdaemon>=3.0->hass-apps) (1.10.1)
Requirement already satisfied: jsonpickle>=1.0 in /usr/lib/python3.8/site-packages (from deepdiff->appdaemon>=3.0->hass-apps) (1.2)
Requirement already satisfied: ordered-set>=3.1.1 in /usr/lib/python3.8/site-packages (from deepdiff->appdaemon>=3.0->hass-apps) (3.1.1)
Requirement already satisfied: certifi>=2017.4.17 in /usr/lib/python3.8/site-packages (from requests>=2.6.0->appdaemon>=3.0->hass-apps) (2019.11.28)
Requirement already satisfied: chardet<3.1.0,>=3.0.2 in /usr/lib/python3.8/site-packages (from requests>=2.6.0->appdaemon>=3.0->hass-apps) (3.0.4)
Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /usr/lib/python3.8/site-packages (from requests>=2.6.0->appdaemon>=3.0->hass-apps) (1.25.7)
Requirement already satisfied: idna<2.9,>=2.5 in /usr/lib/python3.8/site-packages (from requests>=2.6.0->appdaemon>=3.0->hass-apps) (2.8)
Requirement already satisfied: MarkupSafe>=0.23 in /usr/lib/python3.8/site-packages (from Jinja2==2.10.1->appdaemon>=3.0->hass-apps) (1.1.1)
Requirement already satisfied: six in /usr/lib/python3.8/site-packages (from websocket-client->appdaemon>=3.0->hass-apps) (1.14.0)
Requirement already satisfied: async-timeout<4.0,>=3.0 in /usr/lib/python3.8/site-packages (from aiohttp==3.6.2->appdaemon>=3.0->hass-apps) (3.0.1)
Requirement already satisfied: attrs>=17.3.0 in /usr/lib/python3.8/site-packages (from aiohttp==3.6.2->appdaemon>=3.0->hass-apps) (19.3.0)
Requirement already satisfied: multidict<5.0,>=4.5 in /usr/lib/python3.8/site-packages (from aiohttp==3.6.2->appdaemon>=3.0->hass-apps) (4.7.4)
Requirement already satisfied: python-engineio>=3.9.0 in /usr/lib/python3.8/site-packages (from python-socketio->appdaemon>=3.0->hass-apps) (3.11.2)
Requirement already satisfied: cffi>=1.1 in /usr/lib/python3.8/site-packages (from bcrypt==3.1.4->appdaemon>=3.0->hass-apps) (1.13.2)
Requirement already satisfied: pycparser in /usr/lib/python3.8/site-packages (from cffi>=1.1->bcrypt==3.1.4->appdaemon>=3.0->hass-apps) (2.19)
Installing collected packages: observable, cached-property, hass-apps
Successfully installed cached-property-1.5.1 hass-apps-0.20200203.0 observable-1.0.3
WARNING: You are using pip version 19.2.3, however version 20.0.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
[cont-init.d] appdaemon.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[21:23:09] INFO: Starting AppDaemon...
2020-02-04 21:23:11.288103 INFO AppDaemon: AppDaemon Version 4.0.1 starting
2020-02-04 21:23:11.288519 INFO AppDaemon: Python version is 3.8.1
2020-02-04 21:23:11.288966 INFO AppDaemon: Configuration read from: /config/appdaemon/appdaemon.yaml
2020-02-04 21:23:11.289399 INFO AppDaemon: Added log: AppDaemon
2020-02-04 21:23:11.289876 INFO AppDaemon: Added log: Error
2020-02-04 21:23:11.290271 INFO AppDaemon: Added log: Access
2020-02-04 21:23:11.290719 INFO AppDaemon: Added log: Diag
2020-02-04 21:23:11.325941 INFO AppDaemon: Loading Plugin HASS using class HassPlugin from module hassplugin
2020-02-04 21:23:11.380332 INFO HASS: HASS Plugin Initializing
2020-02-04 21:23:11.380858 INFO HASS: HASS Plugin initialization complete
2020-02-04 21:23:11.381969 INFO AppDaemon: Initializing HTTP
2020-02-04 21:23:11.382876 INFO AppDaemon: Using 'ws' for event stream
2020-02-04 21:23:11.405752 INFO AppDaemon: Starting API
2020-02-04 21:23:11.415269 INFO AppDaemon: Starting Admin Interface
2020-02-04 21:23:11.416167 INFO AppDaemon: Starting Dashboards
2020-02-04 21:23:11.439055 INFO HASS: Connected to Home Assistant 0.104.3
2020-02-04 21:23:11.482750 INFO AppDaemon: App 'schedy_coffee' added
2020-02-04 21:23:11.485094 INFO AppDaemon: Found 1 total apps
2020-02-04 21:23:11.486368 INFO AppDaemon: Starting Apps with 1 workers and 1 pins
2020-02-04 21:23:11.488813 INFO AppDaemon: Running on port 5050
2020-02-04 21:23:11.553700 INFO HASS: Evaluating startup conditions
2020-02-04 21:23:11.605058 INFO AppDaemon: Got initial state from namespace default
2020-02-04 21:23:12.179233 INFO AppDaemon: New client Admin Browser connected
2020-02-04 21:23:13.502793 INFO AppDaemon: Scheduler running in realtime
2020-02-04 21:23:13.515958 INFO AppDaemon: Adding /config/appdaemon/apps to module import path
2020-02-04 21:23:13.522225 INFO AppDaemon: Loading App Module: /config/appdaemon/apps/hass_apps_loader.py
2020-02-04 21:23:13.528677 INFO AppDaemon: Initializing app schedy_coffee using class SchedyApp from module hass_apps_loader
2020-02-04 21:23:13.834876 INFO schedy_coffee: *** Welcome to schedy 0.8.0, running on AppDaemon 4.0.1.
2020-02-04 21:23:13.837652 INFO schedy_coffee: *** 
2020-02-04 21:23:13.840322 INFO schedy_coffee: *** This is an app from the hass-apps package.
2020-02-04 21:23:13.842946 INFO schedy_coffee: ***   DOCS: https://hass-apps.readthedocs.io/en/stable/
2020-02-04 21:23:13.845597 INFO schedy_coffee: *** 
2020-02-04 21:23:13.848171 INFO schedy_coffee: *** You like this app, want to honor the effort put into
2020-02-04 21:23:13.850808 INFO schedy_coffee: *** it, ensure continuous development and support?
2020-02-04 21:23:13.853456 INFO schedy_coffee: *** Then please consider making a donation.
2020-02-04 21:23:13.856135 INFO schedy_coffee: ***   DONATE: https://hass-apps.readthedocs.io/en/stable/#donations
2020-02-04 21:23:13.858748 INFO schedy_coffee: *** Thank you very much and enjoy schedy!
2020-02-04 21:23:13.861381 INFO schedy_coffee: *** 
2020-02-04 21:23:13.875453 INFO schedy_coffee: --- Actor type is: 'switch'
2020-02-04 21:23:13.901497 INFO schedy_coffee: --> [R:kitchen] [A:switch.kaffee] Received value of ('off',).
2020-02-04 21:23:14.077699 INFO schedy_coffee: *** Initialization done.
2020-02-04 21:23:14.084492 INFO AppDaemon: App initialization complete`

I appreciate!!
Kind regards

Albert

**Comment by efficiosoft** [comment]
The requirements.txt file is for docker users only. With hassio, you have to specify the URL instead of "hass-apps" in the addon config... You basically just change what you inserted when you set up hass-apps for the first time.

**Comment by efficiosoft** [comment]
It's all described [here](https://hass-apps.readthedocs.io/en/stable/getting-started.html#installation-in-hass-io).

**Comment by adandl** [comment]
Hi,
got it - thanks - still same error-message?

this is what my config looks now:

```
{
  "system_packages": [],
  "python_packages": [
    "https://github.com/efficiosoft/hass-apps/archive/master.zip"
  ],
  "init_commands": []
}
```


```
Requirement already satisfied: ordered-set>=3.1.1 in /usr/lib/python3.8/site-packages (from deepdiff->appdaemon>=3.0->hass-apps==0.20200203.0) (3.1.1)
Requirement already satisfied: jsonpickle>=1.0 in /usr/lib/python3.8/site-packages (from deepdiff->appdaemon>=3.0->hass-apps==0.20200203.0) (1.2)
Requirement already satisfied: MarkupSafe>=0.23 in /usr/lib/python3.8/site-packages (from Jinja2==2.10.1->appdaemon>=3.0->hass-apps==0.20200203.0) (1.1.1)
Requirement already satisfied: multidict>=4.0 in /usr/lib/python3.8/site-packages (from yarl==1.1.0->appdaemon>=3.0->hass-apps==0.20200203.0) (4.7.4)
Requirement already satisfied: async-timeout<4.0,>=3.0 in /usr/lib/python3.8/site-packages (from aiohttp==3.6.2->appdaemon>=3.0->hass-apps==0.20200203.0) (3.0.1)
Requirement already satisfied: attrs>=17.3.0 in /usr/lib/python3.8/site-packages (from aiohttp==3.6.2->appdaemon>=3.0->hass-apps==0.20200203.0) (19.3.0)
Requirement already satisfied: python-engineio>=3.9.0 in /usr/lib/python3.8/site-packages (from python-socketio->appdaemon>=3.0->hass-apps==0.20200203.0) (3.11.2)
Requirement already satisfied: pycparser in /usr/lib/python3.8/site-packages (from cffi>=1.1->bcrypt==3.1.4->appdaemon>=3.0->hass-apps==0.20200203.0) (2.19)
Installing collected packages: cached-property, observable, hass-apps
  Running setup.py install for hass-apps: started
    Running setup.py install for hass-apps: finished with status 'done'
Successfully installed cached-property-1.5.1 hass-apps-0.20200203.0 observable-1.0.3
WARNING: You are using pip version 19.2.3, however version 20.0.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
[cont-init.d] appdaemon.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[21:37:30] INFO: Starting AppDaemon...
2020-02-04 21:37:31.751541 INFO AppDaemon: AppDaemon Version 4.0.1 starting
2020-02-04 21:37:31.752322 INFO AppDaemon: Python version is 3.8.1
2020-02-04 21:37:31.752816 INFO AppDaemon: Configuration read from: /config/appdaemon/appdaemon.yaml
2020-02-04 21:37:31.753311 INFO AppDaemon: Added log: AppDaemon
2020-02-04 21:37:31.753916 INFO AppDaemon: Added log: Error
2020-02-04 21:37:31.754524 INFO AppDaemon: Added log: Access
2020-02-04 21:37:31.755008 INFO AppDaemon: Added log: Diag
2020-02-04 21:37:31.790034 INFO AppDaemon: Loading Plugin HASS using class HassPlugin from module hassplugin
2020-02-04 21:37:31.843580 INFO HASS: HASS Plugin Initializing
2020-02-04 21:37:31.844401 INFO HASS: HASS Plugin initialization complete
2020-02-04 21:37:31.845517 INFO AppDaemon: Initializing HTTP
2020-02-04 21:37:31.846600 INFO AppDaemon: Using 'ws' for event stream
2020-02-04 21:37:31.869114 INFO AppDaemon: Starting API
2020-02-04 21:37:31.878836 INFO AppDaemon: Starting Admin Interface
2020-02-04 21:37:31.879831 INFO AppDaemon: Starting Dashboards
2020-02-04 21:37:31.904452 INFO HASS: Connected to Home Assistant 0.104.3
2020-02-04 21:37:31.956550 INFO AppDaemon: App 'schedy_coffee' added
2020-02-04 21:37:31.958871 INFO AppDaemon: Found 1 total apps
2020-02-04 21:37:31.960040 INFO AppDaemon: Starting Apps with 1 workers and 1 pins
2020-02-04 21:37:31.962231 INFO AppDaemon: Running on port 5050
2020-02-04 21:37:32.011735 INFO HASS: Evaluating startup conditions
2020-02-04 21:37:32.054538 INFO AppDaemon: Got initial state from namespace default
2020-02-04 21:37:33.974636 INFO AppDaemon: Scheduler running in realtime
2020-02-04 21:37:33.985242 INFO AppDaemon: Adding /config/appdaemon/apps to module import path
2020-02-04 21:37:33.992180 INFO AppDaemon: Loading App Module: /config/appdaemon/apps/hass_apps_loader.py
2020-02-04 21:37:34.008684 INFO AppDaemon: Initializing app schedy_coffee using class SchedyApp from module hass_apps_loader
2020-02-04 21:37:34.337832 INFO schedy_coffee: *** Welcome to schedy 0.8.0, running on AppDaemon 4.0.1.
2020-02-04 21:37:34.340435 INFO schedy_coffee: *** 
2020-02-04 21:37:34.343011 INFO schedy_coffee: *** This is an app from the hass-apps package.
2020-02-04 21:37:34.345460 INFO schedy_coffee: ***   DOCS: https://hass-apps.readthedocs.io/en/stable/
2020-02-04 21:37:34.347936 INFO schedy_coffee: *** 
2020-02-04 21:37:34.350365 INFO schedy_coffee: *** You like this app, want to honor the effort put into
2020-02-04 21:37:34.352842 INFO schedy_coffee: *** it, ensure continuous development and support?
2020-02-04 21:37:34.355280 INFO schedy_coffee: *** Then please consider making a donation.
2020-02-04 21:37:34.357744 INFO schedy_coffee: ***   DONATE: https://hass-apps.readthedocs.io/en/stable/#donations
2020-02-04 21:37:34.360188 INFO schedy_coffee: *** Thank you very much and enjoy schedy!
2020-02-04 21:37:34.362790 INFO schedy_coffee: *** 
2020-02-04 21:37:34.376532 INFO schedy_coffee: --- Actor type is: 'switch'
2020-02-04 21:37:34.398723 INFO schedy_coffee: --> [R:kitchen] [A:switch.kaffee] Received value of ('off',).
2020-02-04 21:37:34.540461 INFO schedy_coffee: *** Initialization done.
2020-02-04 21:37:34.544932 INFO AppDaemon: App initialization complete
2020-02-04 21:37:55.795922 INFO AppDaemon: New client Admin Browser connected
2020-02-04 21:38:04.025874 INFO schedy_coffee: --> Attribute 'state' of 'input_boolean.renata' changed from 'off' to 'on', reevaluating <Room R:kitchen>.
2020-02-04 21:38:05.090388 WARNING schedy_coffee: ------------------------------------------------------------
2020-02-04 21:38:05.091347 WARNING schedy_coffee: Unexpected error in worker for App schedy_coffee:
2020-02-04 21:38:05.092273 WARNING schedy_coffee: Worker Ags: {'id': '503e29d405ef4c85b74b8b17b48ed63d', 'name': 'schedy_coffee', 'objectid': '7fa19198093e4c03831473814980ffdb', 'type': 'scheduler', 'function': <function Room.trigger_reevaluation.<locals>._reevaluation_cb at 0xb522ccd0>, 'pin_app': True, 'pin_thread': 0, 'kwargs': {'__thread_id': 'thread-0'}}
2020-02-04 21:38:05.093027 WARNING schedy_coffee: ------------------------------------------------------------
2020-02-04 21:38:05.095731 WARNING schedy_coffee: Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/appdaemon/threading.py", line 766, in worker
    funcref(self.AD.sched.sanitize_timer_kwargs(app, args["kwargs"]))
  File "/usr/lib/python3.8/site-packages/hass_apps/schedy/room.py", line 723, in _reevaluation_cb
    self.apply_schedule(reset=reset)
  File "/usr/lib/python3.8/site-packages/hass_apps/schedy/room.py", line 50, in wrapper
    result = handler(self, *args, **kwargs)
  File "/usr/lib/python3.8/site-packages/hass_apps/schedy/room.py", line 409, in apply_schedule
    self.set_value(new_scheduled_value, force_resend=force_resend)
  File "/usr/lib/python3.8/site-packages/hass_apps/schedy/room.py", line 587, in set_value
    changed |= actor.set_value(value, force_resend=force_resend)[0]
  File "/usr/lib/python3.8/site-packages/hass_apps/schedy/actor/base.py", line 285, in set_value
    self._resending_cb({"left_tries": self.cfg["send_retries"] + 1})
  File "/usr/lib/python3.8/site-packages/hass_apps/schedy/room.py", line 50, in wrapper
    result = handler(self, *args, **kwargs)
  File "/usr/lib/python3.8/site-packages/hass_apps/schedy/actor/base.py", line 93, in _resending_cb
    self.do_send()
  File "/usr/lib/python3.8/site-packages/hass_apps/schedy/actor/generic2.py", line 147, in do_send
    self.app.call_service(service, **data)
  File "/usr/lib/python3.8/site-packages/appdaemon/utils.py", line 191, in inner_sync_wrapper
    f = run_coroutine_threadsafe(self, coro(self, *args, **kwargs))
  File "/usr/lib/python3.8/site-packages/appdaemon/utils.py", line 285, in run_coroutine_threadsafe
    result = future.result(self.AD.internal_function_timeout)
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 439, in result
    return self.__get_result()
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 388, in __get_result
    raise self._exception
  File "/usr/lib/python3.8/site-packages/appdaemon/adapi.py", line 1485, in call_service
    self._check_service(service)
  File "/usr/lib/python3.8/site-packages/appdaemon/adapi.py", line 1369, in _check_service
    raise ValueError("Invalid Service Name: {}".format(service))
ValueError: Invalid Service Name: homeassistant.turn_on
2020-02-04 21:38:05.096556 WARNING schedy_coffee: ------------------------------------------------------------
```

**Comment by bachya** [comment]
@efficiosoft I installed the latest `development` (which is version `0.20200203.0`, I assume) and the error has gone away. 👍 

@adandl That isn't the same error.

**Comment by efficiosoft** [comment]
@bachya Thanks for confirming.

@adandl This one should now be fixed as well. Could you confirm please?

I'll then release a hotfix quickly.

**Comment by adandl** [comment]
Good morning!
All works well!!!
Thank you!!!

**Comment by efficiosoft** [comment]
Ok thanks for confirming. I released Schedy 0.8.1 as a hotfix.

Best regards
Robert
